### PR TITLE
SDFG Breakpoints in Debug Panel

### PR DESCRIPTION
**Pull request by benibenj**
SDFG Breakpoints displayed in Debug Panel
- Left Click -> Opens Viewer, Displays Breakpoints (adds the Overlay), zooms to the state of the breakpoint
- Right Click -> Jump to the C++ file and highlight the matching location
- Click on the cross/remove button -> Removes the Breakpoint on the SDFG

Removed the VSCodeRenderer
- Adding the BreakpointIndicator (Overlay) to the Overlaymanager by registering it

Removed the SDFG Breakpoint View Panel
- Added a button directly in the Viewer to display or hide the Breakpoints (register/deregister BPI overlay)

Note: I will try to reduce the size of the PRs the next times

### CODE REVIEW

File: media/components/sdfv/util/vscode_renderer.js [hunk 1]
```diff
@@ -1,51 +0,0 @@
-// Copyright 2020-2021 ETH Zurich and the DaCe-VSCode authors.
-// All rights reserved.
-
-class VSCodeRenderer extends daceSDFGRenderer {
```
Comment (tbennun) [inline media/components/sdfv/util/vscode_renderer.js:4]: @phschaad do we want to keep this code one way or another?

**Review by tbennun — COMMENTED** [review]
LGTM on my end, apart from the removal of the VSCodeSDFGRenderer extension, which I think is used in other active PRs.

File: media/components/sdfv/util/vscode_renderer.js (hunk 1 above)
Comment (phschaad) [reply media/components/sdfv/util/vscode_renderer.js:4]: I vote yes, I make heavy use of the `VSCodeRenderer` in a currently open PR. I also cleaned up the initialization (see comment above on `index.html`), so I would prefer if we can keep this in - I generally like the minimally-invasive approach of extending the renderer that this enables.

File: media/components/sdfv/index.html [hunk 2]
```diff
@@ -140,18 +144,16 @@ <h5 id="info-title"></h5>
...
             parsed_sdfg = daceParseSDFG(sdfg);
 
             if (daceRenderer) {
                 daceRenderer.set_sdfg(parsed_sdfg);
             } else {
-                daceInitSDFV(null, null, debug_draw);
-                if (parsed_sdfg !== null)
-                    daceRenderer = new VSCodeRenderer(parsed_sdfg, document.getElementById('contents'),
-                        daceMouseEvent, null, debug_draw);
+                daceInitSDFV(parsed_sdfg, null, debug_draw);
```
Comment (phschaad) [inline media/components/sdfv/index.html:156]: This situation gets cleaned up using the `VSCodeRenderer` in a currently open PR from my side as well. I am in favor of keeping the `VSCodeRenderer` in, despite being able to register overlays without it now. There is a comparatively large number of functions in the renderer at this point, that are exclusive and specific to VSCode. They're often guarded by `if (vscode)` clauses etc., but I would prefer if all that functionality would live only inside VSCode. We are discussing a migration of the webclient to an `npm` package, which could further clean this up, but for now I vote on keeping this in. Thoughts?

**Review by phschaad — CHANGES_REQUESTED** [review]
LGTM too, apart from the VSCodeRenderer thing @tbennun correctly pointed out. I left two comments pertaining to that.

File: media/components/sdfv/index.html (hunk 2 above)
Comment (benibenj) [reply media/components/sdfv/index.html:156]: Sounds good to me! I will add the VSCodeRenderer back, but remove the Overlay functionalities which arent needed anymore like on_mouse_event.

**Review by phschaad — APPROVED** [review]
LGTM, thank you!

**Pull request merged**